Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
-                else if (pGameActivity)
-				{
-                    pGameActivity->SetCPUTeam(team);
-				}
-            }
-        }
-    }
-
-	for (int team = Activity::TeamOne; team < Activity::MaxTeamCount; ++team)
-    {
-		// Set up techs
-		GUIListPanel::Item *pTechItem = m_apTeamTechSelect[team]->GetSelectedItem();
-		if (pTechItem)
-		{
-			// If the "random" selection, choose one from the list of loaded techs
-			if (m_apTeamTechSelect[team]->GetSelectedIndex() == 1)//pTechItem->m_ExtraIndex < 0)
-			{
-				int selection = RandomNum<int>(1, m_apTeamTechSelect[team]->GetListPanel()->GetItemList()->size() - 1);
-				m_apTeamTechSelect[team]->SetSelectedIndex(selection);
-				pTechItem = m_apTeamTechSelect[team]->GetSelectedItem();
-
-				// Switch back to -Random-
-				m_apTeamTechSelect[team]->SetSelectedIndex(1);
+
+		char goldString[256];
+		if (m_GoldSlider->GetValue() == m_GoldSlider->GetMaximum()) {
+			std::snprintf(goldString, sizeof(goldString), "Starting Gold: %c Infinite", -58);
+		} else {
+			int startGold = m_GoldSlider->GetValue();
+			startGold = startGold - (startGold % 500);
+			std::snprintf(goldString, sizeof(goldString), "Starting Gold: %c %d oz", -58, startGold);
+		}
+		m_GoldLabel->SetText(goldString);
+
+		for (int teamIndex = Activity::TeamOne; teamIndex < Activity::MaxTeamCount; teamIndex++) {
+			m_TeamAISkillLabel.at(teamIndex)->SetText(Activity::GetAISkillString(m_TeamAISkillSlider.at(teamIndex)->GetValue()));
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioGUI::ClickInPlayerSetup(int clickedPlayer, int clickedTeam) {
+		// Move the player's icon to the correct row.
+		m_PlayerBoxes.at(clickedPlayer).at(clickedTeam)->SetDrawType(GUICollectionBox::Image);
+		const Icon *playerIcon = (clickedPlayer != PlayerColumns::PlayerCPU) ? g_UInputMan.GetSchemeIcon(clickedPlayer) : dynamic_cast<const Icon *>(g_PresetMan.GetEntityPreset("Icon", "Device CPU"));
+		if (playerIcon) { m_PlayerBoxes.at(clickedPlayer).at(clickedTeam)->SetDrawImage(new AllegroBitmap(playerIcon->GetBitmaps32()[0])); }
+
+		// Clear all un-hovered rows of this column.
+		for (int nonHoveredTeam = Activity::TeamOne; nonHoveredTeam < TeamRows::TeamRowCount; ++nonHoveredTeam) {
+			if (nonHoveredTeam != clickedTeam) {
+				m_PlayerBoxes.at(clickedPlayer).at(nonHoveredTeam)->SetDrawType(GUICollectionBox::Color);
+				m_PlayerBoxes.at(clickedPlayer).at(nonHoveredTeam)->SetDrawColor(c_GUIColorBlue);
 			}
+		}
 
-			// Now set the selected tech's module index as what the metaplayer is going to use
-			if (pTechItem)
-				if (pTechItem->m_ExtraIndex == -2)
-					pGameActivity->SetTeamTech(team, "-All-");
-				else
-					pGameActivity->SetTeamTech(team, g_PresetMan.GetDataModuleName(pTechItem->m_ExtraIndex));
+		// If CPU changed team, remove human players from the team.
```

Comment:
I would probably rather these 2 comments be merged into one better one; `Remove human players from their team row if a CPU player is added to it, and vice-versa for CPU players.`